### PR TITLE
Fix simulation testcoverage and make some tests faster

### LINKED ISSUE #776: Improve test coverage in `simulation` submodule
The simulation submodule has poor test coverage, ~20% and we should improve this.

### LINKED ISSUE #836: Docking CI still slow.
The docking CI needs some proper mocking to make sure it runs in a reasonable time.

### PR DESCRIPTION
**Pull request by hmacdope**
## Description
Provide a brief description of the PR's purpose here.
Fixes #836 
FIxes #776 

## Developers certificate of origin
- [X] I certify that this contribution is covered by the MIT license as defined in our [LICENSE](https://github.com/choderalab/asapdiscovery/blob/main/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

### CODE REVIEW

**Comment by hmacdope** [comment]
Docking CI down from 1hr to 20 mins! hooray.

**Comment by hmacdope** [comment]
@jthorton @ijpulidos @chrisiacovella could I grab a review here?

File: asapdiscovery-simulation/asapdiscovery/simulation/tests/test_simulate.py [hunk 1]
```diff
@@ -1,5 +1,15 @@
-import pytest  # noqa: F401
+import pytest
+from asapdiscovery.simulation.simulate import VanillaMDSimulator
```
Comment (chrisiacovella) [inline asapdiscovery-simulation/asapdiscovery/simulation/tests/test_simulate.py:3]: You need to import os here

**Review by chrisiacovella — CHANGES_REQUESTED** [review]
you're missing "import os" in test_simulation statement.

File: asapdiscovery-simulation/asapdiscovery/simulation/tests/test_simulate.py (hunk 1 above)
Comment (hmacdope) [reply asapdiscovery-simulation/asapdiscovery/simulation/tests/test_simulate.py:3]: Ah thanks!

**Comment by hmacdope** [comment]
@chrisiacovella should be fixed now :)

**Comment by jthorton** [comment]
Great work @hmacdope glad to see the testing speeds improved. It looks like we can get a big coverage boost by testing the rmsd restraint code in simulate I have no idea how often we use that but it seems important. Can you have a go at adding some tests for it? Feel free to make it another PR if the testing is more complicated.

**Comment by hmacdope** [comment]
@jthorton I have added tests for the RMSD code, and fixed a bug along the way (this is why we test ), would you mind reviewing?

**Review by jthorton — APPROVED** [review]
LGTM!

**Comment by hmacdope** [comment]
I am going to assume that was a runner issue.

**Pull request merged**